File: Desenvolvimento/server/src/ApplicationCore/Entities/Empresa.cs
using ApplicationCore.Interfaces.Base;
using System.Collections.Generic;

namespace ApplicationCore.Entities
{
    public class Empresa: IBaseEntity<long>
    {
        public Empresa() { }

        public long Id { get; set; }
        public string Sigla { get; set; }
        public string Nome { get; set; }
        public bool Ativo { get; set; }

        public virtual ICollection<Setor> Setores { get; set; } = new List<Setor>();
        public virtual ICollection<Assunto> Assuntos { get; set; } = new List<Assunto>();
        public virtual ICollection<EmpresaEndereco> EmpresaEnderecos { get; set; } = new List<EmpresaEndereco>();
    }
}

File: Desenvolvimento/client/src/@fuse/types/models/empresa.ts
import { Setor } from './setor';
import { BaseModel } from '../base-model';
export class Empresa extends BaseModel {
    constructor(
        public id: number,
        public sigla?: string,
        public nome?: string,
        public ativo?: boolean,
        public setores?: Array<Setor>,
    ) { super(); }
}